api/src/main/java/jakarta/data/page/Pagination.java

```diff
@@ -22,7 +22,7 @@
 /**
  * Built-in implementation of PageRequest.
  */
-record Pagination(long page, int size, Mode mode, Cursor type, boolean requestTotal) implements PageRequest {
+record Pagination(long page, int size, Mode mode, Cursor type, boolean requestTotal) implements PageRequest, CursoredPageRequest, OffsetBasedPageRequest {
```

Comment:
I would split this into `CursoredPageRequestRecord` and `OffsetBasedPageRequestRecord`.

Reply:
I like this.